File: 2026-01/src/day_31/boj_15652/Main.java
package day_31.boj_15652;

import java.io.*;
import java.util.*;

public class Main {

    static StringBuilder sb = new StringBuilder();
    static int n, m;
    static int[] sel;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        sel = new int[m];

        dfs(1, 0);
        System.out.println(sb);
    }

    static void dfs(int idx, int selIdx) {
        if (selIdx == m) {
            for (int x : sel) {
                sb.append(x).append(" ");
            }
            sb.append("\n");
            return;
        }

        for (int i = idx; i <= n; i++) {
            sel[selIdx] = i;
            dfs(i, selIdx + 1);
        }
    }
}

File: 2026-01/src/day_31/boj_15652/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int n, m;
int sel[8];

void dfs(int idx, int selIdx) {
    if (selIdx == m) {
        for (int i = 0; i < m; i++) {
            cout << sel[i] << ' ';
        }
        cout << '\n';
        return;
    }

    for (int i = idx; i <= n; i++) {
        sel[selIdx] = i;
        dfs(i, selIdx + 1);
    }
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n >> m;

    dfs(1, 0);
}
